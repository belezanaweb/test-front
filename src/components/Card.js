import React, { Fragment } from 'react'; 
import styled from 'styled-components'; 

const Title = styled.div`
    height: 17px;
    width: 319px;
    color: #999;
    font-size: 14px;
    font-weight: 700;
    line-height: 17px;
    padding-top: 2vh;
    padding-bottom: 2vh;
`

const Box = styled.div`
    width: 341px;
    border-radius: 3px;
    background-color: #FFF;
    box-shadow: 1px 1px 5px 0 rgba(0,0,29,0.22);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-evenly;
`

const Card = (props) => {
    return (
        <Fragment>
            <Title>
                {props.title}
            </Title>
            <Box>
                {props.children}
            </Box> 
        </Fragment>

    )
}

export default Card;