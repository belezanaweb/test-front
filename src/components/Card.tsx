import { PropsWithChildren } from 'react'

function Card({ children }: PropsWithChildren) {
  return (
    <section className="rounded m-3 m-5 px-2 pt-5 pb-7 w-11/12 max-w-md bg-white border border-neutral-300">
      {children}
    </section>
  )
}

export default Card
