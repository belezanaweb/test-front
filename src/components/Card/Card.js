import React from 'react'
import { createUseStyles } from 'react-jss'

const useStyles = createUseStyles({
  defaultCard: {
    width: '80%',
    minWidth: '320px',
    maxWidth: '600px',
    minHeight: '20px',
    borderRadius: '3px',
    backgroundColor: '#ffffff',
    boxShadow: '2px 2px 2px 1px rgba(0, 0, 0, 0.1)',
    border: 'none',
    padding: '10px',
    boxSizing: 'border-box',
    margin: '20px auto'
  }
})

const Card = (props) => {
  const { text, click } = props
  const classes = useStyles()

  return <div className={classes.defaultCard}>{props.children}</div>
}

export default Card
