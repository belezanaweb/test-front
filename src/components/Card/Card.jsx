import clsx from 'clsx';
import useStyles from './useStyles';

const Card = ({ children, className }) => {
  const classes = useStyles();

  return <div className={clsx(classes.root, className)}>{children}</div>;
};

export default Card;
