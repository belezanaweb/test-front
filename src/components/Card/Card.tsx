import styled from 'styled-components';

import Box from '../Box';

const Card = styled(Box)`
  background-color: #FFF;
  border-radius: 3px;
  box-shadow: 1px 1px 5px 0 rgba(0, 0, 29, 0.22);
  padding: 12px;
`;

export default Card;
