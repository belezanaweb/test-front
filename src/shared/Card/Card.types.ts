export type CardProps = {
  centered?: boolean
}

export type StyledCardProps = {
  $centered?: CardProps['centered']
}
