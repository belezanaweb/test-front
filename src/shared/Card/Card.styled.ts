import styled from 'styled-components'
import { CardProps } from './Card.types';

export const CardContainer = styled.div<CardProps>`
  background: #fff;
  border: 1px solid #ccc;
  border-radius: 4px;
  padding: 15px 12px;
  ${({ centered }) => centered &&`
    text-align: center;
  `}
`;