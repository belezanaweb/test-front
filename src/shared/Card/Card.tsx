import { PropsWithChildren } from 'react'

import { CardContainer } from './Card.styled'
import { CardProps } from './Card.types'

export const Card = ({ children, centered }: PropsWithChildren<CardProps>) => {
  return (
    <CardContainer centered={centered}>
      {children}
    </CardContainer>
  )
}