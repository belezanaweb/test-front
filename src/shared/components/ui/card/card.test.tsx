import React from 'react';
import { render, screen } from '../../../../../config/test-utils';

import Card from './index';

const mock = {
  title: "L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium - Máscara de Reconstrução 500g",
  price: "R$ 225,90",
  image:
    'https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/1/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png',
}

describe('<Card />', () => {
  it('should render component with <article> tag', () => {
    render(<Card {...mock} />, {});
    expect(screen.getByRole('article'));
  })

  it('should render component with default props and match snapshot', () => {
    const { container } = render(<Card {...mock} />, {});
    expect(container.firstChild).toMatchSnapshot();
  })
})
