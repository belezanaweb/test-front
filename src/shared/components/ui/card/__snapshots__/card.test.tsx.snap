// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render component with default props and match snapshot 1`] = `
.c0 {
  background: #fff;
  border: 0.1rem solid #eee;
  border-radius: 0.3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 1.5rem;
  min-height: 9rem;
  padding: 1rem;
}

.c0:first-child {
  margin-top: 0;
}

.c1 {
  height: 6.5rem;
  width: 6.5rem;
}

.c2 {
  height: 6.5rem;
  width: 6.5rem;
}

.c3 {
  padding-bottom: 2rem;
  padding-left: 1rem;
  position: relative;
}

.c4 {
  color: #212121;
  font-size: 1.3rem;
  margin-bottom: 1.5rem;
}

.c5 {
  color: #212121;
  font-size: 1.4rem;
  font-weight: 700;
  position: absolute;
  bottom: 0.4rem;
  right: 1rem;
}

<article
  class="c0"
>
  <figure
    class="c1"
  >
    <img
      alt="L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium - Máscara de Reconstrução 500g"
      class="c2"
      src="https://res.cloudinary.com/beleza-na-web/image/upload/f_auto,fl_progressive,q_auto:best/v1/imagens/1/loreal-professionnel-expert-absolut-repair-cortex-lipidium-mascara-de-reconstrucao-500g-24410-963234120108391775.png"
    />
  </figure>
  <dl
    class="c3"
  >
    <dt
      class="c4"
    >
      L'Oréal Professionnel Expert Absolut Repair Cortex Lipidium - Máscara de Reconstrução 500g
    </dt>
    <dd
      class="c5"
    >
      R$ 225,90
    </dd>
  </dl>
</article>
`;
